frontend/src/pages/clusterSettings/ClusterSettings.tsx

```diff
@@ -53,6 +58,9 @@ const ClusterSettings: React.FC = () => {
   const [defaultDeploymentMode, setDefaultDeploymentMode] = React.useState<DeploymentMode>(
     defaultSingleModelDeploymentMode,
   );
+  const [instructLabDSCState, setInstructLabDSCState] = React.useState(isInstructLabEnabled(dsc));
```

Comment:
this wont work always becuase dsc will be undefined on load until it is fetched

Reply:
also. this soesnt need to be a state. you are only updating when you submit which you dont need to do anyway.

what you should do is just have something like

```tsx
const [dsc, , dscError] = useDsc();
const instructLabDSCEnabled = isInstructLabEnabled(dsc)
```